__tests__/utils.test.ts

```diff
@@ -0,0 +1,88 @@
+// ./__tests__/utils.test.ts
+//
+// Unittests for the L-Dist utility functions.
+
+// Vitest essentail imports.
+import { describe, expect, it } from 'vitest';
+
+// Local utility functions import.
+import {
+  calculateScore,
+  getMatchingIndices,
+  getMaxLevenshteinDistance,
+  levenshteinFullMatrixSearch,
+} from '../src/utils';
+
+// levenshteinFullMatrixSearch test suite.
+describe('levenshteinFullMatrixSearch', () => {
+  it('returns a matrix of the correct dimensions', () => {
+    const query = 'hello';
+    const target = 'world';
+    const matrix = levenshteinFullMatrixSearch(query, target);
+    expect(matrix.length).toBe(query.length + 1);
+    expect(matrix[0].length).toBe(target.length + 1);
+  });
+
+  it('returns a matrix with the correct values', () => {
+    const query = 'hello';
+    const target = 'world';
+    const matrix = levenshteinFullMatrixSearch(query, target);
+    expect(matrix[0][0]).toBe(0);
+    expect(matrix[0][1]).toBe(1);
+    expect(matrix[1][0]).toBe(1);
+    expect(matrix[1][1]).toBe(1);
+    expect(matrix[1][2]).toBe(2);
+    expect(matrix[5][5]).toBe(4);
```

Comment:
same , inline snapshot will be better, or even regular snapshot if its too big

Reply:
@ad1992

- [x] Replaced all hard-coded `toBe` with `toMatchSnapshot` in 9bfc05c.